test/scale-infra/deploy_ks_core.yaml

```diff
@@ -0,0 +1,114 @@
+             sudo apt-get install apt-transport-https --yes
+             echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | sudo tee /etc/apt/sources.list.d/helm-stable-debian.list
+             sudo apt-get update
+             sudo apt-get install helm
+
+    - name: Register a variable, ignore errors and continue
+      ansible.builtin.command: kubectl config use-context kscore
+      register: result
+      ignore_errors: true
+
+    - name: install ingress controller
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore
+             kubectl --context kscore label node --all ingress-ready="true"
+             kubectl --context kscore apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-0.25.1/example/kind-nginx-ingress-with-SSL-passthrough.yaml
+
+    - name: Delete old KS deployment
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore --ignore-not-found
+             helm delete ks-core  --ignore-not-found
+             helm -n open-cluster-management delete status-addon  --ignore-not-found
+             kubectl --context kscore delete ns open-cluster-management-hub open-cluster-management  --ignore-not-found
+             kubectl config unset contexts.wds1 --ignore-not-found
+             kubectl --context kscore delete -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-"{{ ks_release }}"/example/kind-nginx-ingress-with-SSL-passthrough.yaml  --ignore-not-found
+      when: result is success
+
+    - name: install KubeStellar using core helm-cart
+      shell: |
+             export KUBESTELLAR_VERSION="{{ ks_release }}"
+             helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
+             --version $KUBESTELLAR_VERSION \
+             --set-json='ITSes=[{"name":"its1","type":"host"}]' \
+             --set-json='WDSes=[{"name":"wds1"}]' \
+             --set-json='verbosity.default=5'
+
+    - name: Get kubeconfig "context" for wds1 space
+      shell: |
+             kflex ctx wds1
+
+    - name: update wds1 space with the correct port for the ingress controller
+      shell: kubectl --context kscore -n ingress-nginx get svc ingress-nginx-controller -o=jsonpath='{.spec.ports[1].nodePort}'
+      register: ingress_port
+
+    # - name: Debugging message
+    #   ansible.builtin.debug:
+    #    msg: "{{ ingress_port.stdout }}"
+
+    - name: Replace old hostname with new hostname (requires Ansible >= 2.4)
+      ansible.builtin.replace:
+        path: /root/.kube/config
+        regexp: '(\s+)https://wds1\.localtest\.me:9443(\s+.*)?$'
+        replace: '\1https://wds1.localtest.me:{{ ingress_port.stdout }}\2'
+
+
+    - name: Sleep for 30 seconds and continue with play
```

Comment:
Why is this wait here? Experience with the demo env creation script showed that there is actually wide variation in how much time anything takes; waiting for the relevant state condition is the only reliable wait.

Reply:
Oh, I removed that waiting - it is no longer needed. I was using it to help with some debugging.